tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/QueueWorkerBackgroundService.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services
+{
+    internal abstract class QueueWorkerBackgroundService : BackgroundService
+    {
+        private const string ActivitySourceName = "Azure.Sdk.Tools.PipelineWitness.Queue";
+        private static readonly ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+
+        private readonly ILogger logger;
+        private readonly QueueClient queueClient;
+        private readonly QueueClient poisonQueueClient;
+        private readonly TelemetryClient telemetryClient;
+        private readonly IOptions<PipelineWitnessSettings> options;
+
+        public QueueWorkerBackgroundService(
+            ILogger logger,
+            QueueClient queueClient,
+            QueueClient poisonQueueClient,
+            TelemetryClient telemetryClient,
+            IOptions<PipelineWitnessSettings> options)
+        {
+            this.logger = logger;
+            this.telemetryClient = telemetryClient;
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            this.queueClient = queueClient;
+            this.poisonQueueClient = poisonQueueClient;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            await this.queueClient.CreateIfNotExistsAsync();
+            await this.poisonQueueClient.CreateIfNotExistsAsync();
+
+            this.logger.LogInformation("Starting ExecuteAsync {TypeName}", this.GetType().Name);
```

Comment:
Might be more useful to have this statement above the create calls.

Reply:
agreed